src/CloudGraphsDFG/services/CommonFunctions.jl

```diff
@@ -226,13 +253,108 @@ $(SIGNATURES)
 Get the Neo4j labels for any node instance.
 """
 function _getLabelsForInst(dfg::CloudGraphsDFG,
-                            inst::Union{User, Robot, Session, N, APPE, ABDE}; parentKey::Union{Nothing, Symbol}=nothing)::Vector{String} where
+                            inst::Union{User, Robot, Session, VariableNodeData, N, APPE, ABDE};
+                            parentKey::Union{Nothing, Symbol}=nothing)::Vector{String} where
                             {N <: DFGNode, APPE <: AbstractPointParametricEst, ABDE <: AbstractBigDataEntry}
     labels = _getLabelsForType(dfg, typeof(inst), parentKey=parentKey)
     typeof(inst) <: DFGVariable && push!(labels, String(getLabel(inst)))
     typeof(inst) <: DFGFactor && push!(labels, String(getLabel(inst)))
     typeof(inst) <: AbstractPointParametricEst && push!(labels, String(inst.solverKey))
+    typeof(inst) <: VariableNodeData && push!(labels, String(inst.solverKey))
     typeof(inst) <: AbstractBigDataEntry && push!(labels, String(inst.key))
-    # Some are children of nodes, so add that in if it's set.
     return labels
 end
+
+## Common CRUD calls for subnode types (PPEs, VariableSolverData, BigData)
+
+function _listVarSubnodesForType(dfg::CloudGraphsDFG, variablekey::Symbol, dfgType::Type, keyToReturn::String; currentTransaction::Union{Nothing, Neo4j.Transaction}=nothing)::Vector{Symbol}
+    query = "match (subnode:$(join(_getLabelsForType(dfg, dfgType, parentKey=variablekey),':'))) return subnode.$keyToReturn"
+    @debug "[Query] _listVarSubnodesForType query:\r\n$query"
+    result = nothing
+    if currentTransaction != nothing
+        result = currentTransaction(query; submit=true)
+    else
+        tx = transaction(dfg.neo4jInstance.connection)
+        tx(query)
+        result = commit(tx)
+    end
+    length(result.errors) > 0 && error(string(result.errors))
+    vals = map(d -> d["row"][1], result.results[1]["data"])
+    return Symbol.(vals)
+end
+
+function _getVarSubnodeProperties(dfg::CloudGraphsDFG, variablekey::Symbol, dfgType::Type, nodeKey::Symbol; currentTransaction::Union{Nothing, Neo4j.Transaction}=nothing)
+    query = "match (subnode:$(join(_getLabelsForType(dfg, dfgType, parentKey=variablekey),':')):$nodeKey) return properties(subnode)"
+    @debug "[Query] _getVarSubnodeProperties query:\r\n$query"
+    result = nothing
+    if currentTransaction != nothing
+        result = currentTransaction(query; submit=true)
+    else
+        tx = transaction(dfg.neo4jInstance.connection)
+        tx(query)
+        result = commit(tx)
+    end
+    length(result.errors) > 0 && error(string(result.errors))
+    length(result.results[1]["data"]) != 1 && error("Cannot find subnode '$nodeKey' for variable '$variablekey'")
+    length(result.results[1]["data"][1]["row"]) != 1 && error("Cannot find subnode '$nodeKey' for variable '$variablekey'")
+    return result.results[1]["data"][1]["row"][1]
+end
+
+function _matchmergeVarSubnode!(
```

Comment:
I think we should push for verbNoun compliance, i.e. `_matchmergeVariableSubnode!`.

Another cool thing is we have not formally defined the underscore `_verb` yet -- how about formal definition:
```
Underscore _verb* is a reserved convention for internal functions and variables 
that may not necessarily be useful to a user who is not interested in the internal 
developer aspects of the source code.  These functions, however, still follow
the general verbNoun convention as far possible.
```

Reply:
cc @Affie

Reply:
Agreed, will update it.  

Also - underscore functions are not exported.

Reply:
added definition to verbNoun documentation wiki